protocol/messenger.go

```diff
@@ -966,6 +966,19 @@ func (m *Messenger) LeaveGroupChat(ctx context.Context, chatID string) (*Messeng
 }
 
 func (m *Messenger) saveChat(chat *Chat) error {
+	_, ok := m.allChats[chat.ID]
+	if !ok && chat.Active {
+		if err := m.Join(*chat); err != nil {
```

Comment:
It seems like `saveChat` has more than one responsibility here. Why do we need to join here?

Reply:
we can be explicit about when to join a chat, but basically every time we save a new chat we want to join it (if active).
We can also split it, but basically we will be duplicating a bit of code, in at least 3 places: 

- when status-react calls `SaveChat`->`Join`

- when we receive a one to one message `SaveChat`->`Join`

- when we receive a `SyncPublicChatMessage` `SaveChat`->`Join` 

So I felt having it always is probably safer (I can't see a case where you would want to `Save` a chat but not join if active), but happy to go with anything to be honest

Reply:
So maybe we can have a public method `SaveAndJoin()` which is implemented as:
```
func (m *Messenger) SaveAndJoin(chat *Chat) error {
    if err := m.saveChat(chat); err != nil {
        return err
    }
    return m.join(chat)
}

func (m *Messenger) Join(chat Chat) error {
    return m.join(chat)
}
```

Reply:
Actually, I have removed that bit, it's not needed as of now as filters are managed explicitly by status-react.

Once we go around moving filters management in status-go, we might re-think and go for something as you described.